### ISSUE TITLE
DPE Inspector: Variables set via component on Script Canvas component are duplicated and have no identifier

### ISSUE THREAD

**Issue by jckand-amzn** [o3de/o3de#15905]
**Describe the bug**
Variables set via component on Script Canvas component are duplicated and have no identifier

**Steps to reproduce**
1. Launch the Editor with the DPE enabled in the AutomatedTesting project:
```
Editor.exe --ed_enableDPE=true
```
2. Open a new or existing level.
3. Create a new entity with a Script Canvas component.
4. Open Script Canvas.
5. Create a new graph with an EntityId variable that is set from the component.
6. Add an On Graph Start node, and connect it to a Get <EntityIdVariable> node.
7. Save the graph.
8. Expand the Variables section to assign the EntityId variable via the entity picker.

**Expected behavior**
Variable is exposed only once.

**Actual behavior**
Variable is exposed twice, and can be set to different values. The variable is also unlabled, so multiple variables can be confusing.

**Screenshots/Video**
![image](https://user-images.githubusercontent.com/82226555/236930907-c8d9cb91-a549-4880-96a9-c868ecf4110f.png)

**Found in Branch**
development

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
42a9733b187a5c72616e450a506e91a08d18829b

**Additional context**
1. The display of the Variables/Unused Variables/Dependencies also differs from the RPE. They act like containers that can be added to at will rather than pulling values from the assigned graph.

**Comment by LB-ArturZieba** [comment]
We have verified on Development (c9db420) that viewing a Script Canvas component with a graph assigned that contains a variable exposed to the component while DPE is enabled causes the Editor to crash. Should this be logged as a separate issue, or will fix of this issue potentially solve the crash?

**Comment by LB-ArturZieba** [comment]
Issue verified as fixed on the Stabilization/2310 (0be3e09) build.

**Comment by LB-ArturZieba** [comment]
Reopening the issue. On Stabilization/2310 (6f4ea1d) branch the Editor crashes when viewing a Script Canvas component with a graph assigned that contains a variable exposed to the component while DPE is enabled.

**Comment by cgalvan** [comment]
There was a new issue filed for the same crash behavior (https://github.com/o3de/o3de/issues/16762), so I'm going to re-close this issue and resolve the issue against the other GHI.
